main.c

```diff
@@ -466,28 +512,21 @@ int main(void) {
   chSysInit();
 
   profiles[currentProfile].callback(ledColors, ledIntensity);
+  updateAnimationSpeed();
 
   // Setup masks to all be 0xFF at the start
-  for (size_t i = 0; i < KEY_COUNT; ++i)
-  {
-    ledMasks[i] = 0xFF;
-  }
+  memset(ledMasks, 0xFF, sizeof(ledMasks));
 
   palClearLine(LINE_LED_PWR);
   sdStart(&SD1, &usart1Config);
 
   // Setup Column Multiplex Timer
   gptStart(&GPTD_BFTM0, &bftm0Config);
-  gptStartContinuous(&GPTD_BFTM0, 1);
-
-  // Setup Animation Timer
-  gptStart(&GPTD_BFTM1, &lightAnimationConfig);
+  gptStartContinuous(&GPTD_BFTM0, SERIAL_CHECK_FREQUENCY);
 
+  // idle loop, copied form lib/chibios/os/rt/src/chsys.c
   while (true) {
```

Comment:
Not sure if necessary to put this, but seems to be not making any harm.
Copied from [here](https://github.com/qmk/ChibiOS/blob/313416b8fda90d9973a749a0a35970956852c286/os/rt/src/chsys.c#L72-L83)

Reply:
This is good to have. wait_for_interrupt calls the `WFI` instruction. which puts the arm to a low power state. It should help with battery, but in the presence of those badly designed LEDs. I don't know how much this helps. I assume you know how the LEDs on this board has no current limiting settings, like seriously not even a resistor. It is literally 5V -> PMOS -> LED -> NMOS -> GND.

Reply:
Nope, did not know this. Thanks for explaining.

Reply:
Yeah this is why we try to do the sPWM and all that stuff so it don't light up too long. Basically we are hoping we turn it on before the current ramp up to the max we turn it off.